# Cleansing Data Youtube Rewind 2018

#### Import Library

In [1]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [2]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [3]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2018"
collection_name = "cleansing_2018"

#### Import dataset dari MongoDB

In [4]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [5]:
df

_id                 channelId      videoId  \
0       6651606257713e470060633e  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
1       6651606257713e470060633f  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
2       6651606257713e4700606340  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
3       6651606257713e4700606341  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
4       6651606257713e4700606342  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
...                          ...                       ...          ...   
111459  6651619157713e47006216a1  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
111460  6651619157713e47006216a2  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
111461  6651619157713e47006216a3  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
111462  6651619157713e47006216a4  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   
111463  6651619157713e47006216a5  UC-z1ubmS-iHMVr83EEwRS2g  mCyITaDib7M   

                                              textDisplay  \
0                                                  Hellow   
1                                       Kangenn Dehh.. 🥶😭   
2                                             Dari 2024 😄   
3                               Baru nonton di tahun 2024   
4                                  2018 ternyata 6th lalu   
...                                                   ...   
111459                          setelah lama menanti wkwk   
111460                                             mantap   
111461                                              Aloha   
111462                                 PINK GUY - セックス大好き   
111463  SMA SANTA MARIA-Pekanbaru selalu memberikan ya...   

                                             textOriginal   authorDisplayName  \
0                                                  Hellow     @azkaranuri6676   
1                                       Kangenn Dehh.. 🥶😭     @BLUEBERRY_ICE.   
2                                             Dari 2024 😄       @Justkhanza__   
3                               Baru nonton di tahun 2024     @aryafajrin4802   
4                                  2018 ternyata 6th lalu             @clyzea   
...                                                   ...                 ...   
111459                          setelah lama menanti wkwk     @erlanggasp5492   
111460                                             mantap          @rianr7336   
111461                                              Aloha          @nick.8697   
111462                                 PINK GUY - セックス大好き  @azaryapranata1108   
111463  SMA SANTA MARIA-Pekanbaru selalu memberikan ya...    @mishaeljehoshua   

                                    authorProfileImageUrl  \
0       https://yt3.ggpht.com/A0rqQbMrRkVy7KLsE6h9tf0_...   
1       https://yt3.ggpht.com/hJaeCYyM3zkG27NAwy7ns5LT...   
2       https://yt3.ggpht.com/6SUOHpxm8zO3JNtlcQRN0VHv...   
3       https://yt3.ggpht.com/ytc/AIdro_lzu5d-GRupA7fc...   
4       https://yt3.ggpht.com/e7d9_hgTBHJoCEpP_WgnB-SH...   
...                                                   ...   
111459  https://yt3.ggpht.com/ytc/AIdro_kPkXrD0JhY2huK...   
111460  https://yt3.ggpht.com/ytc/AIdro_lDYfVzt_oCeQFG...   
111461  https://yt3.ggpht.com/RrBhVBns4BCdpe785raxfCtM...   
111462  https://yt3.ggpht.com/ytc/AIdro_k_cCFPZI6jfrqr...   
111463  https://yt3.ggpht.com/ppU9UKoTkEeWvS_8JSPAindV...   

                                 authorChannelUrl  \
0          http://www.youtube.com/@azkaranuri6676   
1          http://www.youtube.com/@BLUEBERRY_ICE.   
2            http://www.youtube.com/@Justkhanza__   
3          http://www.youtube.com/@aryafajrin4802   
4                  http://www.youtube.com/@clyzea   
...                                           ...   
111459     http://www.youtube.com/@erlanggasp5492   
111460          http://www.youtube.com/@rianr7336   
111461          http://www.youtube.com/@nick.8697   
111462  http://www.youtube.com/@azaryapranata1108   
111463    http://www.youtube.com/@mishaeljehoshua   

                              authorChannelId  canRa

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(101952, 14)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(92038, 14)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan textOriginal

In [8]:
df = df[['_id', 'textOriginal']]
df

_id  \
0       6651606257713e470060633e   
1       6651606257713e470060633f   
2       6651606257713e4700606340   
3       6651606257713e4700606341   
4       6651606257713e4700606342   
...                          ...   
111453  6651619157713e470062169b   
111454  6651619157713e470062169c   
111457  6651619157713e470062169f   
111458  6651619157713e47006216a0   
111459  6651619157713e47006216a1   

                                             textOriginal  
0                                                  Hellow  
1                                       Kangenn Dehh.. 🥶😭  
2                                             Dari 2024 😄  
3                               Baru nonton di tahun 2024  
4                                  2018 ternyata 6th lalu  
...                                                   ...  
111453                                         Yuhuuuuuuu  
111454                                         Mantulll:)  
111457                              Di like dlu pasti nya  
111458  Yt rewind indo 2018 sangat keren !!!\nSangat b...  
111459                          setelah lama menanti wkwk  

[92038 rows x 2 columns]

#### Menghapus nilai yang kosong

In [9]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id             0
textOriginal    0
dtype: int64

Dari yang awalnya berjumlah 111464 row berkurang menjadi 92038 row

#### Mengimpan Clean Comments ke MongoDB Local

In [10]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.
